### PR TITLE
Update sudoku module: adding solve functionality

### LINKED ISSUE #4: Adding solver to sudoku module
Implement a solver

### PR DESCRIPTION
**Pull request by JimOgren**
Jag har ersatt `model.py` med `sudoku.py` som nu också har funktion för att lösa alla möjliga sudoku. 

De flesta funktioner från tidigare finns kvar men några onödiga har tagits bort. Valideringen sker i två steg. Först `validate_sudoku_input` som bara kontrollerar att det är en `List[int]` som innehåller (0-9) och är av korrekt längd. Den andra delen handlar om sudokun är lösbar och det kollas egentligen bara genom att testa lösa den. 

Lösningen sker i två strategier: *constraint propagation* och *depth-first search*. Constraint propagation liknar som en människa löser en sudoku: om en siffra på en ruta utesluter möjligheten att samma siffra finns på någon ruta i samma rad, kolumn eller box ("peers"). För svåra sudoku är detta inte tillräckligt och då görs en depth-first rekursiv sökning. Om lösningen någon gång misslyckas returneras en tom lista i annat fall returneras lösningen som `List[int]`.

Jag har uppdaterat test. Säkert behövs en del edge cases fortfarande kontrolleras. I `post_validate` kollas nu bara `validate_sudoku_input` och inte om den går att lösa. Det kan vi lägga till senare. Closes #4

### CODE REVIEW

File: server/app_sudoku/domain/sudoku.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        peers[index] = peers_list
+    return squares, units, peers
+
+
+SQUARES, UNITS, PEERS = generate_squares_units_peers()
+
+
+# --------------- above part of init --------------------
+# --------------- below part of class --------------------
+
```
Comment (JimOgren) [inline server/app_sudoku/domain/sudoku.py:67]: Tanken är att funktioner ovanför bara behövs en gång vid en initialisering. Funktioner nedan kan vara metoder till sudoku-klassen.

**Comment by taleldayekh** [comment]
Nice! Jag tar en titt imorgon, hann inte idag. :)

**Review by taleldayekh — APPROVED** [review]
Go for it 👍🏼
